components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml

```diff
@@ -4693,6 +4693,177 @@ document.observe("xwiki:dom:loaded", function() {
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators> ,|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage=Always on this page|onDemand=On demand|always=Always on this wiki</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>default</cache>
+    </property>
+    <property>
+      <code>var PhenoTips = (function(PhenoTips) {
+  var widgets = PhenoTips.widgets = PhenoTips.widgets || {};
+  widgets.SegmentedBar = Class.create({
+    DEFAULT_SEGMENTS : 5,
+    DEFAULT_FILL : '#777',
+    DEFAULT_BORDER : '#777',
+    DEFAULT_SEGMENT_HEIGHT : '5px',
+    DEFAULT_DISPLAY_PERCENT_VALUE_FLAG : true,
+    /**
+    A PhenoTips widget for creating a segment bar html element. Example:
+
+        var barOptions = {
+          segments : 10,
+          fill: '#625',
+          border : '#987',
+          segmentHeight : '10px',
+          segmentWidth : '3px',
+          displayPercentValueFlag : false
+        };
+        container.insert(new PhenoTips.widgets.SegmentedBar(score, barOptions).generateSegmentedBar());
+
+    @param value The value that this bar is meant to display. Must be between 0 and 1.
+
+    @param options Options for styling the segmented bar, all values are optional:
+
+    {
+      segments: The number of segments in the bar. Default is 5.,
+      fill: The fill color for the segments. Default is #777.,
+      border: The border color of the segments. Default is #000.,
+      segmentHeight: The height of each individual segment. Default is 5px.,
+      segmentWidth: The width of each individual segment. Default is equal to either the specified height or the default height.,
+      displayPercentValueFlag: Option to display the percentage value after the bar. Default is true
+    }
+
+    */
+    initialize : function(value, options) {
+      if (value &gt; 1 || value &lt; 0) {
+        console.log("Invalid segmented bar value");
+      }
+      this.value = value;
+      this.options = options || {};
+      this.segments = this.options.segments || this.DEFAULT_SEGMENTS;
+      this.fill = this.options.fill || this.DEFAULT_FILL;
+      this.border = this.options.border || this.DEFAULT_BORDER;
+      this.segmentHeight = this.options.segmentHeight || this.DEFAULT_SEGMENT_HEIGHT;
+      this.segmentWidth = (this.options.segmentWidth || this.options.segmentHeight) || this.DEFAULT_SEGMENT_HEIGHT;
+      this.displayPercentValueFlag = (this.options.displayPercentValue != undefined) ? this.options.displayPercentValue : this.DEFAULT_DISPLAY_PERCENT_VALUE_FLAG;
+    },
+    generateSegmentedBar: function() {
+      var bar = new Element('div', {
+        'class' : 'segmented-bar',
+        'style' : 'color: ' + this.border + ';',
```

Comment:
Instead of setting the style inline as an attribute, you can call [`setStyle({...})`](http://api.prototypejs.org/dom/Element/prototype/setStyle/), which has the advantage that it accepts an explicit map instead of one string.